sklearn/cluster/_hierarchical_fast.pxd

```diff
@@ -0,0 +1,9 @@
+from ..utils._typedefs cimport ITYPE_t
+
+cdef class UnionFind:
+    cdef ITYPE_t next_label
+    cdef ITYPE_t[:] parent
+    cdef ITYPE_t[:] size
+
+    cdef void union(self, ITYPE_t m, ITYPE_t n)
+    cdef ITYPE_t fast_find(self, ITYPE_t n)
```

Comment:
This change, along with that of `_hierarchical_fast.pxd` are just to allow `_linkage.pyx` to reuse the `UnionFind` class that they share in common, rather than duplicating code.